cloud_storage_handler/utils.py

```diff
@@ -0,0 +1,37 @@
+"""Utility functions for the Cloud Storage Handler package."""
+
+import os
+from pathlib import Path
+
+from foca import Foca
+
+from cloud_storage_handler.custom_config import CustomConfig
+from cloud_storage_handler.exceptions import ConfigNotFoundError
+
+
+def get_config_path() -> Path:
+    """Get the configuration path.
+
+    Returns:
+      The path of the config file.
+    """
+    # Determine the configuration path
+    if config_path_env := os.getenv("CSH_FOCA_CONFIG_PATH"):
+        return Path(config_path_env).resolve()
+    else:
+        return (Path(__file__).parents[1] / "deployment" / "config.yaml").resolve()
```

Comment:
If you let FOCA handle the config, you probably won't need this.

Reply:
This is similar to what is in TESK, unless FOCA gives us a way to export an env to configure path of config.yaml, I think this is easiest way which could be easy removed once FOCA has that func.

But I also think this should be applications job not FOCA's as if someone want to run multiple FOCA application in one shell session it wont work as all would defaul to the same env variable, ie this will force user to run the application in different shell session, but if application configures it, they can have different env variable and the shell env conflict wont occur.

Thoughts?

Reply:
As discussed elsewhere, I think there's probably a way to namespace env vars to a given service instance (check `env_prefix` in [Pydantic Settings docs](https://docs.pydantic.dev/latest/concepts/pydantic_settings/#usage)) to address your concern in your second point.

As for how to do it best: I'm not a big fan of instantiating a new app for the config, from a "code philosphy" point of view, but also (and mostly) because it's dangerous: If the config ever changes between consecutive reads, this could lead to really weird inconsistencies. And then there's the problem with having to somewhere define the location of the config file...

So I feel that that there should be just one clear place and mechanism to read the config, such that the service is (re)configured only upon startup OR when it is explicitly requested to do so.

So my suggestion would be to try solutions in the following order:
1. *Avoid:* The FOCA config is available in app context (via `current_app`), and so will be available for all request-related work. It will _also_ be available whenever you have access to the actual FOCA app instance, so for things that you need to do between instantiating the instance and starting the service, you should have access to all config params as well, in principle (you just may have to pass the app instance if you call a function, e.g., to set up the MinIO client). These scenarios should cover the vast majority of use cases, and where they don't, first consider redesigning your approach so that they _do_ fit.
2. *Manually push app context*: If you really cannot avoid a situation where you need access to the config, but you neither can't get access to the app instance, nor to `current_app` in app context, try to push the app context manually (see Flask documentation for that).
3. *Instantiate a new FOCA app instance*: Only if all else fails, instantiate a new FOCA app instance to read the config. But document the risk this poses (config inconsistency) and gracefully handle the problem of finding the config (it may not always be at the same place).